### PR TITLE
feat(core): `PrimitiveCheckbox` full customization through DI

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Comment by waterplea** [comment]
@fynnfeldpausch check this one out :)

**Comment by fynnfeldpausch** [comment]
> @fynnfeldpausch check this one out :)

Nice. I really like the icons concept now. It is much more understandable than overriding a dictionary. Looking forward to bringing this concept to all the other components. Will probable start helping out next week

File: projects/core/components/primitive-checkbox/checkbox-options.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import {InjectionToken} from '@angular/core';
+import {TuiContextWithImplicit} from '@taiga-ui/cdk';
+import {TuiAppearance} from '@taiga-ui/core/enums';
+import {TuiSizeL} from '@taiga-ui/core/types';
+import {PolymorpheusContent} from '@tinkoff/ng-polymorpheus';
+
+export interface CheckboxOptions {
+    readonly size: TuiSizeL;
+    readonly appearances: {
```
Comment (MarsiBarsi) [inline projects/core/components/primitive-checkbox/checkbox-options.ts:9]: you can also use `readonly appearances: Readonly<{ ... }>

File: projects/core/tokens/checkbox-appearance.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import {InjectionToken} from '@angular/core';
 import {TuiAppearance} from '@taiga-ui/core/enums';
```
Comment (MarsiBarsi) [inline projects/core/tokens/checkbox-appearance.ts:2]: Remove this one

Comment (MarsiBarsi) [reply projects/core/tokens/checkbox-appearance.ts:2]: and do not forget to update demo ;)

**Review by MarsiBarsi — APPROVED** [review]

**Pull request merged**